_posts/selinux-vs-apparmor.md

```diff
@@ -0,0 +1,126 @@
+
+```
+$ getenforce
+Disabled
+```
+
+SELinux is installed but not running, you will need to activate and reboot:
+
+```
+$ selinux-activate
+$ reboot
+```
+
+This should put SELinux into Permissive mode. To check, run `sestatus` and you should get the following:
+
+```
+SELinux status:                 enabled
+SELinuxfs mount:                /sys/fs/selinux
+SELinux root directory:         /etc/selinux
+Loaded policy name:             default
+Current mode:                   permissive
+Mode from config file:          permissive
+Policy MLS status:              enabled
+Policy deny_unknown status:     allowed
+Max kernel policy version:      30
+```
+
+## Setting Modes
+
+Run: 
+
+```
+$ audit2why -al
+[...]
+```
+This will return would-be blocked operations since your last boot. It is tedious but worth going through as SELinux has many reported bugs. We would suggest going through each bug one-by-one and follow [this Red Hat guide](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/security-enhanced_linux/sect-security-enhanced_linux-fixing_problems-allowing_access_audit2allow) for changing access using the `audit2allow` command.
+
+If you are not confident, **we strongly suggest testing this on a virtual machine**.
+
+You can now change the mode to enforcing:
+ * to test, run: `setenforce = 1` in your terminal
+ * or if you are confident, add `enforcing = 1` to the kernel command line in your `/etc/default/grub` file
+
+and then reboot as earlier. You will now be running SELinux's default policy. To change this, simply manoeuvre back to the root user and set it back to permissive via `setenforce = 0`. 
+
+Provided you have access to the root user, you can feel free to play around with SELinux's [other policies](https://docs.fedoraproject.org/en-US/quick-docs/changing-selinux-states-and-modes/) until you are comfortable.
+
+## Finishing Up
+
+And that concludes our intro to SELinux. Play around with policies until you are comfortable, then you can ship SELinux to your own servers with confidence. SELinux essentially acts as a sandbox protecting your information from fradulent or corrupted access by external programs or daemons. As for AppArmor, it's just another MAC protocol that's used particularly for Ubuntu. It's easier to use as it deals with pathing rules; but SELinux is widely considered the more secure of the two. SELinux is better for those who are very familiar with Unix based systems, but AppArmor is another great introduction to MAC. 
+
+SELinux is a great way to implement security, but it is known for its bugs and disruptive mechanisms. Actual sandboxing is another alternative to protecting your kernel.
+
+Another way you can look to secure your devices- particularly if they are edge-based or use the net; is our very own agent here at WoTT. It's simple to set up and flexible, so why not head on over to our [getting started]({{site.url}}/documentation/getting-started) page and give it a go?
```

Comment:
Replace last paragraph with:

Good work getting to the end of this tutorial. You have taken a big step in securing your linux environment. There are additional steps you can take to improve security posture like CVE scanning, firewall management and service configuration. [WoTT](https://wott.io/documentation/getting-started) can help you do that with a lightweight agent that scans for vulnerabilities and provides clear recommended actions to fix those issues.